src/main/scala/io/iohk/ethereum/blockchain/BlocksRepository.scala

```diff
@@ -0,0 +1,68 @@
+package io.iohk.ethereum.blockchain
+
+import akka.util.ByteString
+import io.iohk.ethereum.db.storage.{BlockBodiesStorage, BlockHeadersStorage, BlockNumberMappingStorage}
+import io.iohk.ethereum.domain.Block
+
+/**
+  * This interface presents an API to persist and query blocks within the blockchain.
+  */
+trait BlocksRepository {
+  def blockHeadersStorage: BlockHeadersStorage
+  def blockBodiesStorage: BlockBodiesStorage
+  def blockNumberMappingStorage: BlockNumberMappingStorage
+
+  /**
+    * Allows to query for a block based on it's hash
+    *
+    * @param hash Keccak hash of the block that's being searched
+    * @return Block if found
+    */
+  def getBlockByHash(hash: ByteString): Option[Block] =
```

Comment:
protocol messages allow only to ask for block header / block body I think we should add API for loading only header/body